src/Components/GEOSldas_GridComp/GEOSlandassim_GridComp/clsm_ensupd_upd_routines.F90

```diff
@@ -4403,12 +4465,199 @@ subroutine cat_enkf_increments(                               &
           
        end do
 
+       ! ----------------------------------       
+       
+    case (11) select_update_type  ! 1d snow analysis (Toure et al. 2018 empirical gain); snow cover fraction obs
+       
+       if (logit) write (logunit, *) 'get 1d snow increments (Toure et al. 2018 empirical gain); snow cover fraction obs'
+       
+       ! ensure that max SWE increment parameter is less than WEMIN; larger increments make no sense because
+       ! at SWE=WEMIN, the tile is fully snow covered (asnow=1)
+
+       if (SCF_ANA_MAXINCRSWE>WEMIN)  call ldas_abort(LDAS_GENERIC_ERROR, Iam, 'must use SCF_ANA_MAXINCRSWE<=WEMIN')
+       
+       ! get target for snow layer thickness (as used in Catchment over land tiles)
+       
+       targetthick(1)        = CATCH_DZ1MAX
+       
+       targetthick(2:N_snow) = 1./(N_snow-1.)
+       
+       ! identify the obs species of interest       
+
+       N_select_varnames  = 1      
+
+       select_varnames(1) = 'asnow'
+  
+       call get_select_species(                                           &
+            N_select_varnames, select_varnames(1:N_select_varnames),      &
+            N_obs_param, obs_param, N_select_species, select_species )
+       
+       allocate(select_tilenum(1))
+       
+       swe_incr = 0.   ! total SWE increment; initialize to NO CHANGE
+       
+       ! loop through tiles and compute increments
+       
+       do kk=1,N_catd
+          
+          ! find observations for tile kk
+          
+          select_tilenum(1) = l2f(kk)
+          
+          call get_ind_obs(                                           &
+               N_obs,            Observations,                        &
+               1,                select_tilenum,                      &
+               N_select_species, select_species(1:N_select_species),  &
+               N_selected_obs,   ind_obs )
+          
+          if (N_selected_obs > 0) then
+             
+             ! NEED TO ALLOW FOR MULTIPLE asnow OBS (I.E., FROM MULTIPLE SENSORS, SAY, MODIS AND VIIRS)
+             !
+             ! if statement added to maintain 0-diff for now; can be removed later
+             ! (that is, compute average even when N_selected_obs=1)
+             
+             if (N_selected_obs > 1) then    
+                
+                ! compute average obs value
+                
+                tmp_obs = sum(Observations(ind_obs(1:N_selected_obs))%obs)
+                
+                tmp_obs = tmp_obs/real(N_selected_obs)
+             
+             else
+                
+                tmp_obs = Observations(ind_obs(1))%obs
+                
+             end if
```

Comment:
If statement introduced in the hope of maintaining 0-diff.  Needs cleanup later.